### PR TITLE
fix coverage unit test

### LINKED ISSUE #1010: Add executable_lines metric for SQ 6.2 and later
Starting with SQ6.2 there is the new executable_lines metric to calculate the code coverage.
See also: https://groups.google.com/forum/#!topic/sonarqube/_B0Tnz3TyUA

### LINKED ISSUE #1156: CxxCoverageSensor not able to get the covergae property sonar.cxx.coverage.reportPath
Using plugin sonar-cxx 0.9.7 with SonarQube Version 6.3 (build 19869), I was not able to get the coverage in the SonarQube. It just show 0% all time. Same configuration with same xml file generated with gcov/gcovr shows a valid coverage % in SonarQube Version 4.5.7. 

The configuration is:

```
# we want this set explicitly
sonar.cxx.coverage.forceZeroCoverage=True

# paths to the reports
sonar.cxx.coverage.reportPath=/sonar/coverage.xml
sonar.cxx.xunit.reportPath=sonar/xunit-reports/xunit-reports.xml
sonar.cxx.xunit.provideDetails=True
sonar.cxx.cppcheck.reportPath=sonar/cppcheck-report.xml
sonar.cxx.rats.reportPath=sonar/rats-report.xml
```

And the log from SonarQube 6.3 says (See the part related to CxxCoverageSensor):

```
00:06:54.131 17:02:02.805 INFO: Sensor CxxRatsSensor [cxx] (done) | time=1145ms
00:06:54.131 17:02:02.805 INFO: Sensor CxxXunitSensor [cxx]
00:06:54.131 17:02:02.805 DEBUG: Runs unit test import sensor only at top level project skip : Module Key = 'king:phoenix-toolbelt-sonar:phoenix-social-facebook'
00:06:54.131 17:02:02.805 INFO: Sensor CxxXunitSensor [cxx] (done) | time=0ms

00:06:54.131 17:02:02.805 INFO: Sensor CxxCoverageSensor [cxx]
00:06:54.131 17:02:02.805 INFO: Searching reports by relative path with basedir '/workspace/toolbelt_osx_sonar/mobile' and search prop ''
00:06:54.131 17:02:02.805 INFO: Undefined report path value for key ''
00:06:54.131 17:02:02.805 INFO: Issues processed = 0
00:06:54.131 17:02:02.805 INFO: Sensor CxxCoverageSensor [cxx] (done) | time=0ms

00:06:54.131 17:02:02.805 INFO: Sensor CxxCppCheckSensor [cxx]
00:06:54.131 17:02:02.805 INFO: Searching reports by relative path with basedir '/workspace/toolbelt_osx_sonar/mobile' and search prop 'sonar.cxx.cppcheck.reportPath'
00:06:54.131 17:02:02.805 DEBUG: Normalized report includes to '[/workspace/toolbelt_osx_sonar/mobile/sonar/cppcheck-report.xml]'
00:06:54.131 17:02:02.806 INFO: Scanner found '1' report files
00:06:54.131 17:02:02.806 INFO: Parser will parse '1' report files
00:06:54.131 17:02:02.806 INFO: Processing report '/workspace/toolbelt_osx_sonar/mobile/sonar/cppcheck-report.xml'
00:06:54.131 17:02:02.806 DEBUG: Parsing 'Cppcheck V2' format
00:06:54.131 17:02:02.807 DEBUG: Parsing 'Cppcheck V1' format
```

While other properties as cppcheck are found and executed, the one from the coverage is not found and is not processed.

Any idea about this?

### PR DESCRIPTION
**Pull request by Bertk**
- not implemented FORCE_ZERO_COVERAGE_KEY
- fix bullseye coverage regression
- close #1156
- close #1010

---
This change is [Reviewable](https://reviewable.io/reviews/sonaropencommunity/sonar-cxx/1189)

### CODE REVIEW

**Comment by guwirth** [comment]
@Bertk without looking to you changes:
* think the topic is a little bit more complicated, starting with SQ6.2 #1010 is necessary
* don't know with which SQ version: sonar.cxx.coverage.itReportPath and sonar.cxx.coverage.overallReportPath are no more supported
* in general: think we need a warning checking the config file for no more or deprecated settings and maybe give hints what the replacement is. Think would be good to to this on one central place?

**Comment by Bertk** [comment]
@guwirth: travis CI fails because  `Recovery is disabled` and it should fail. How can this be fixed? 

```
INFO: Sensor c++ XunitSensor [cxx]
INFO: Scanner found '1' report files
INFO: Parser will parse '1' report files
INFO: Processing report '/home/travis/build/SonarOpenCommunity/sonar-cxx/integration-tests/testdata/googletest_project/invalid_report.xml'
ERROR: Cannot feed the data into SonarQube, details: 'com.ctc.wstx.exc.WstxUnexpectedCharException: Unexpected character 't' (code 116) in prolog; expected '<'
 at [row,col {unknown-source}]: [1,1]'
INFO: Recovery is disabled, failing analysis : 'com.ctc.wstx.exc.WstxUnexpectedCharException: Unexpected character 't' (code 116) in prolog; expected '<'
 at [row,col {unknown-source}]: [1,1]'
INFO: ------------------------------------------------------------------------
INFO: EXECUTION FAILURE
INFO: ------------------------------------------------------------------------
INFO: Total time: 11.523s
INFO: Final Memory: 47M/311M
INFO: ------------------------------------------------------------------------
ERROR: Error during SonarQube Scanner execution
java.lang.IllegalStateException: Unexpected character 't' (code 116) in prolog; expected '<'
 at [row,col {unknown-source}]: [1,1]
	at org.sonar.cxx.sensors.utils.CxxUtils.validateRecovery(CxxUtils.java:122)
	at org.sonar.cxx.sensors.tests.xunit.CxxXunitSensor.execute(CxxXunitSensor.java:105)
```

**Comment by guwirth** [comment]
@Bertk without looking into the details: in case you like to turn error recovery on for one integration test forward the setting on the command line
```
 @SqApi56 @SqApi62
   Scenario: googletest report is invalid
       GIVEN the project "googletest_project"
       WHEN I run "sonar-scanner -Dsonar.cxx.errorRecoveryEnabled=true/false ... 
```

**Comment by Bertk** [comment]
@guwirth  Sorry Günter, but I never touched travis or appveyor before. Could you please check this.

**Comment by guwirth** [comment]
> @guwirth  ... Could you please check this.

@Bertk can have a look this evening...

**Comment by Bertk** [comment]
@guwirth Hi Günter, for me this looks like a travis reliability issue because appveyor works fine. I execute within eclipse and ```mvn test``` with any errors.
```
-------------------------------------------------------------------------------
Test set: org.sonar.cxx.sensors.coverage.CxxBullseyeCoverageSensorTest
-------------------------------------------------------------------------------
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.063 sec - in org.sonar.cxx.sensors.coverage.CxxBullseyeCoverageSensorTest
```

Error while test is executed in travis CI:
```
org.sonar.cxx.sensors.coverage.CxxBullseyeCoverageSensorTest
shouldReportCorrectCoverage(org.sonar.cxx.sensors.coverage.CxxBullseyeCoverageSensorTest)  Time elapsed: 0.074 sec  <<< FAILURE!
org.junit.ComparisonFailure: expected:<1> but was:<null>
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.fest.assertions.ConstructorInvoker.newInstance(ConstructorInvoker.java:36)
	at org.fest.assertions.ComparisonFailureFactory.newComparisonFailure(ComparisonFailureFactory.java:60)
	at org.fest.assertions.ComparisonFailureFactory.comparisonFailure(ComparisonFailureFactory.java:46)
	at org.fest.assertions.Fail.comparisonFailed(Fail.java:83)
	at org.fest.assertions.Fail.failIfNotEqual(Fail.java:71)
	at org.fest.assertions.GenericAssert.isEqualTo(GenericAssert.java:217)
	at org.fest.assertions.IntAssert.isEqualTo(IntAssert.java:61)
	at org.sonar.cxx.sensors.coverage.CxxBullseyeCoverageSensorTest.shouldReportCorrectCoverage(CxxBullseyeCoverageSensorTest.java:84)
```

**Comment by guwirth** [comment]
@Bertk there are many differences between Travis and AppVeyor:
* AppVeyor:
  * testing only with SQ5.6
  * testing on Windows system
* Travis:
  * testing with SQ5.6 to SQ6.4
  * testing on Linux system

In the past we had only timeout issues on Travis and no logical issues. Problems came typically from different file systems and naming conventions (e.g. slash, backslash, ...).

Looking to CxxBullseyeCoverageSensorTest there are some tests handling the differences (some are commented out). So question is: what's different on Linux?

**Comment by Bertk** [comment]
@guwirth Thank you Günter. There are multiple issues with the travis tests 

- CxxBullseyeCoverageSensorTest fails on all SQ versions with travis : might be a related to Linux runtime environment
- the following scenarios fail with SQ 6.2 and later versions
```
integration-tests/features/importing_coverage.feature:40  Importing coverage reports
integration-tests/features/importing_coverage.feature:98  Importing coverage reports zeroing coverage for untouched files
integration-tests/features/importing_coverage.feature:164  Zeroing coverage measures without importing reports
integration-tests/features/smoketest.feature:64  Smoketest
integration-tests/features/test_create_rules_and_bullseye_import.feature:8  GoogleTestWithBullseyeAndVsProject
```

Unfortunately I do not have a Linux environment to debug CxxBullseyeCoverageSensorTest and the other issue is a bug in CxxCoverageSensor which I have to fix.

File: cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java [hunk 1]
```diff
@@ -72,42 +90,76 @@ public CxxCoverageSensor(CxxCoverageCache cache, CxxLanguage language) {
...
+      if (context.settings().hasKey(FORCE_ZERO_COVERAGE_KEY)) {
+        LOG.warn("Property '{}' is deprecated and its value will be ignored.", FORCE_ZERO_COVERAGE_KEY);
+      }
+
+      Set<File> reportPaths = getReportPaths(context);
+      if (reportPaths.isEmpty()) {
+        return;
+      }
+      
       LOG.debug("Parsing coverage reports");
```
Comment (guwirth) [inline cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java]: LOG.isDebugEnabled missing

File: cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java [hunk 2]
```diff
@@ -72,42 +90,76 @@ public CxxCoverageSensor(CxxCoverageCache cache, CxxLanguage language) {
...
-      overallCoverageMeasures = processReports(context, overallReports, this.cache.overallCoverageCache());
-      saveMeasures(context, overallCoverageMeasures, CoverageType.OVERALL);
-    }
-    
-    if (this.language.getBooleanOption(FORCE_ZERO_COVERAGE_KEY)) {
-      LOG.debug("Zeroing coverage information for untouched files");
-      zeroMeasuresWithoutReports(context, coverageMeasures, itCoverageMeasures, overallCoverageMeasures, linesOfCode);
+    } else {
+      if (this.language.hasKey(REPORT_PATH_KEY)) {
+        LOG.debug("Parsing coverage reports");
```
Comment (guwirth) [inline cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java]: LOG.isDebugEnabled missing

File: cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java [hunk 3]
```diff
@@ -72,42 +90,76 @@ public CxxCoverageSensor(CxxCoverageCache cache, CxxLanguage language) {
...
+        List<File> reports = getReports(this.language, context.fileSystem().baseDir(), REPORT_PATH_KEY);
+        coverageMeasures = processReports(context, reports, this.cache.unitCoverageCache());
+        saveMeasures(context, coverageMeasures, CoverageType.UNIT);
+      }
+  
+      if (this.language.hasKey(IT_REPORT_PATH_KEY)) {
+        LOG.debug("Parsing integration test coverage reports");
```
Comment (guwirth) [inline cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java]: LOG.isDebugEnabled missing

File: cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java [hunk 4]
```diff
@@ -72,42 +90,76 @@ public CxxCoverageSensor(CxxCoverageCache cache, CxxLanguage language) {
...
+        List<File> itReports = getReports(this.language, context.fileSystem().baseDir(), IT_REPORT_PATH_KEY);
+        itCoverageMeasures = processReports(context, itReports, this.cache.integrationCoverageCache());
+        saveMeasures(context, itCoverageMeasures, CoverageType.IT);
+      }
+  
+      if (this.language.hasKey(OVERALL_REPORT_PATH_KEY)) {
+        LOG.debug("Parsing overall test coverage reports");
```
Comment (guwirth) [inline cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java]: LOG.isDebugEnabled missing

File: cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java [hunk 5]
```diff
@@ -174,24 +232,27 @@ private void saveZeroValueForResource(InputFile inputFile, SensorContext context
...
+                      report.getAbsolutePath(), cacheCov.size());
+              }
+              cacheCov.put(report.getAbsolutePath(), measuresTotal);
+              // Only use first coverage parser with handles the data correctly
+              break;
+            }
           }
+          measuresTotal.putAll(cacheCov.get(report.getAbsolutePath()));
+        } catch (EmptyReportException e) {
+          LOG.debug("Report is empty {}", e);
```
Comment (guwirth) [inline cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java]: LOG.isDebugEnabled missing

File: cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java (hunk 1 above)
Comment (guwirth) [inline cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java]: LOG.isDebugEnabled missing

File: cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java [hunk 6]
```diff
@@ -36,33 +37,50 @@
...
+  @Deprecated
   public static final String OVERALL_REPORT_PATH_KEY = "coverage.overallReportPath";
+  @Deprecated
   public static final String FORCE_ZERO_COVERAGE_KEY = "coverage.forceZeroCoverage";
-  
+
+  // Configuration properties for SQ 6.2
+  public static final Version SQ_6_2 = Version.create(6, 2);
+  private boolean isSQ_6_2_or_newer;
+  public static final String REPORT_PATHS_KEY = "coverage.reportPaths";
```
Comment (guwirth) [inline cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java]: should be `reportsPaths` to be aligned with vstest/nunit

File: cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java [hunk 7]
```diff
@@ -72,42 +90,76 @@ public CxxCoverageSensor(CxxCoverageCache cache, CxxLanguage language) {
...
     Map<String, CoverageMeasures> overallCoverageMeasures = null;
 
-    LOG.debug("Coverage BaseDir '{}' ", context.fileSystem().baseDir());    
-    
-    if (this.language.hasKey(REPORT_PATH_KEY)) {
+    LOG.info("Coverage BaseDir '{}' ", context.fileSystem().baseDir());
+
+    if (isSQ_6_2_or_newer) {
+      if (context.settings().hasKey(FORCE_ZERO_COVERAGE_KEY)) {
+        LOG.warn("Property '{}' is deprecated and its value will be ignored.", FORCE_ZERO_COVERAGE_KEY);
```
Comment (guwirth) [inline cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java]: Would prefer to do this at one central place in the plugin:
- xxx no more supported use yyy instead
- xxx is deprecated and its value will be ignored

File: cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java [hunk 8]
```diff
@@ -282,13 +344,63 @@ private void checkLineCoverage(NewCoverage newCoverage, CoverageMeasure measure)
...
+        } else {
+          LOG.info("Add coverage report: '{}' '{}'", reportPath, report.getAbsolutePath());
+          reportPaths.add(report);
+        }
+      }
+    }
+
+    if (settings.hasKey(REPORT_PATH_KEY)) {
+      warnUsageOfDeprecatedProperty(settings, REPORT_PATH_KEY);
+      File report = fs.resolvePath(settings.getString(REPORT_PATH_KEY));
```
Comment (guwirth) [inline cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java]: not sure but think is also a comma separated list and not only a single file?

File: cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java [hunk 9]
```diff
@@ -282,13 +344,63 @@ private void checkLineCoverage(NewCoverage newCoverage, CoverageMeasure measure)
...
+      if (!report.isFile()) {
+        LOG.info("Coverage UT report not found: '{}'", settings.getString(REPORT_PATH_KEY));
+      } else {
+        reportPaths.add(report);
+      }
+    }
+
+    if (settings.hasKey(IT_REPORT_PATH_KEY)) {
+      warnUsageOfDeprecatedProperty(settings, IT_REPORT_PATH_KEY);
+      File report = fs.resolvePath(settings.getString(IT_REPORT_PATH_KEY));
```
Comment (guwirth) [inline cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java]: not sure but think is also a comma separated list and not only a single file?

File: cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java [hunk 10]
```diff
@@ -282,13 +344,63 @@ private void checkLineCoverage(NewCoverage newCoverage, CoverageMeasure measure)
...
+  }
+  
   @Override
   protected String getSensorKey() {
     return KEY;
   }  
 
   @Override
   protected String reportPathKey() {
+    if (isSQ_6_2_or_newer) {
```
Comment (guwirth) [inline cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java]: Would change this for all versions to new naming. See no reason why we should be backward compatible as long there is a warning message.

File: cxx-sensors/src/main/java/org/sonar/cxx/sensors/utils/CxxReportSensor.java [hunk 11]
```diff
@@ -252,7 +247,7 @@ private void saveViolation(SensorContext sensorContext, String ruleRepoKey,
     // handles file="" situation -- file level
     if ((filename != null) && (!filename.isEmpty())) {
       String root = sensorContext.fileSystem().baseDir().getAbsolutePath();
-      String normalPath = CxxUtils.normalizePathFull(filename, root);
+      String normalPath = CxxUtils.normalizePathFull(filename.replaceAll("\\\\", "/"), root);
```
Comment (guwirth) [inline cxx-sensors/src/main/java/org/sonar/cxx/sensors/utils/CxxReportSensor.java]: maybe we should use https://github.com/SonarSource/sonarqube/blob/master/sonar-plugin-api/src/main/java/org/sonar/api/utils/PathUtils.java?

File: cxx-squid/src/main/java/org/sonar/cxx/visitors/CxxFileLinesVisitor.java [hunk 12]
```diff
@@ -71,6 +89,9 @@ public void visitToken(Token token) {
       String[] tokenLines = token.getValue().split("\n", -1);
       for (int line = token.getLine(); line < token.getLine() + tokenLines.length; line++) {
         linesOfCode.add(line);
+//        if (!token.isGeneratedCode()) {
```
Comment (guwirth) [inline cxx-squid/src/main/java/org/sonar/cxx/visitors/CxxFileLinesVisitor.java]: do we need this isGeneratedCode?

**Comment by guwirth** [comment]
@Bertk had yesterday a look to the integration tests why they are no more working:
* First problem is the renaming of the configuration property `coverage.reportPath` to `coverage.reportPaths`. This new property is not available in the integration projects. Would recommend to set it back to `coverage.reportPath`. Would rename all `reportPath` in one later step to `reportsPaths`. That's the reason why the metrics are null.
* Second problem is that SQ62 API does no more support `sonar.cxx.coverage.itReportPath` and `sonar.cxx.coverage.overallReportPath`. These properties have to be removed from SQ62 tests.

Like to explain how this Travis tests are working that you can fix it:
* Configuration is in the file .travis.yml. Here are the different SQ version listed. To handle different API versions there is also `SONARAPI` defined (https://github.com/SonarOpenCommunity/sonar-cxx/blob/master/.travis.yml#L6). As you can see API is SqApi56 for SQ 5.6 to SQ6.1 and then SqApi62.
* Travis is using the .feature files to run the tests. In the files over `Scenario` there is always the label `@SqApi56`, `@SqApi62` or both of them. A scenario is always only running if this label is defined for a SQ version. In case a scenario behaves in the same way for both versions both labels are defined. In case of different parameter to handover or different metric results a scenario must be duplicated and adapted according to the SQ behaviour.
* I saw issues in two .feature files:
   * https://github.com/SonarOpenCommunity/sonar-cxx/blob/master/integration-tests/features/importing_coverage.feature
   * https://github.com/SonarOpenCommunity/sonar-cxx/blob/master/integration-tests/features/smoketest.feature
* After renaming the property to 'reportPath' you should get metric values again. Remove sonar.cxx.coverage.itReportPath` and `sonar.cxx.coverage.overallReportPath` from **coverage tests** label with ``@SqApi62`. In case the metric values are different validate them and adapt it in the `@SqApi62` scenario.
* Scenario: Smoketest
  * would not remove `sonar.cxx.coverage.itReportPath` and `sonar.cxx.coverage.overallReportPath`
  * would ignore deprecated warnings instead in `AND the analysis log contains no error/warning messages except those matching:`

**Comment by guwirth** [comment]
@Bertk do yo need more help?

File: cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java (hunk 6 above)
Comment (Bertk) [inline cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java]: can you please more precise? what should be aligned?

File: cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java (hunk 1 above)
Comment (Bertk) [inline cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java]: OK.

Comment (Bertk) [inline cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java]: OK.

File: cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java (hunk 5 above)
Comment (Bertk) [inline cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java]: I think this is OK within the catch block and performance is not relevant here.

File: cxx-squid/src/main/java/org/sonar/cxx/visitors/CxxFileLinesVisitor.java (hunk 12 above)
Comment (Bertk) [inline cxx-squid/src/main/java/org/sonar/cxx/visitors/CxxFileLinesVisitor.java]: Maybe not but I did not see any coverage result until now. Do you know how functional macros are handled by the coverage tools?

File: cxx-sensors/src/main/java/org/sonar/cxx/sensors/utils/CxxReportSensor.java (hunk 11 above)
Comment (Bertk) [inline cxx-sensors/src/main/java/org/sonar/cxx/sensors/utils/CxxReportSensor.java]: OK. I will use ```PathUtils.sanitize(filename)``` for now.

**Comment by Bertk** [comment]
@guwirth Hi Günter, do you have any idea what is the problem with travis?

**Comment by guwirth** [comment]
@Bertk had yesterday a look to the integration tests why they are no more working:
* First problem is the renaming of the configuration property `coverage.reportPath` to `coverage.reportPaths`. This new property is not available in the integration projects. Would recommend to set it back to `coverage.reportPath`. Would rename all `reportPath` in one later step to `reportsPaths`. That's the reason why the metrics are null.
* Second problem is that SQ62 API does no more support `sonar.cxx.coverage.itReportPath` and `sonar.cxx.coverage.overallReportPath`. These properties have to be removed from SQ62 tests.

Like to explain how this Travis tests are working that you can fix it:
* Configuration is in the file .travis.yml. Here are the different SQ version listed. To handle different API versions there is also `SONARAPI` defined (https://github.com/SonarOpenCommunity/sonar-cxx/blob/master/.travis.yml#L6). As you can see API is SqApi56 for SQ 5.6 to SQ6.1 and then SqApi62.
* Travis is using the .feature files to run the tests. In the files over `Scenario` there is always the label `@SqApi56`, `@SqApi62` or both of them. A scenario is always only running if this label is defined for a SQ version. In case a scenario behaves in the same way for both versions both labels are defined. In case of different parameter to handover or different metric results a scenario must be duplicated and adapted according to the SQ behaviour.
* I saw issues in two .feature files:
   * https://github.com/SonarOpenCommunity/sonar-cxx/blob/master/integration-tests/features/importing_coverage.feature
   * https://github.com/SonarOpenCommunity/sonar-cxx/blob/master/integration-tests/features/smoketest.feature
* After renaming the property to 'reportPath' you should get metric values again. Remove `sonar.cxx.coverage.itReportPath` and `sonar.cxx.coverage.overallReportPath` from **coverage tests** label with `@SqApi62`. In case the metric values are different validate them and adapt it in the `@SqApi62` scenario.
* Scenario: Smoketest
  * would not remove `sonar.cxx.coverage.itReportPath` and `sonar.cxx.coverage.overallReportPath`
  * would ignore deprecated warnings instead in `AND the analysis log contains no error/warning messages except those matching:`

File: cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java (hunk 6 above)
Comment (guwirth) [inline cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/CxxCoverageSensor.java]: @Bertk we have now ReportPath, ReportPatths and your new ReportPaths. Please use only `ReportPath`. In a final step we should change everything from ReportPath => ReportPatths (but only after everything is working again). Here you see the different flavours :https://github.com/SonarOpenCommunity/sonar-cxx/wiki/Supported-configuration-properties.

**Comment by Bertk** [comment]
@guwirth Since yesterday travis service is not working : ```The command "sudo apt-get -qq update" failed 3 times.``` Do you know how to fix this?

**Comment by guwirth** [comment]
@Bertk please squash and try again. Can't download Boost package. Restarted this and also other open pull request. In other one it works?

**Comment by guwirth** [comment]
@Bertk after rebase I have the same problem now. Will create a Travis ticket tomorrow.

**Comment by Bertk** [comment]
@guwirth maybe you can reduce the number of active SQ versions as well. SQ 6.0, 6.1, 6.2 and 6.3.1 are somewhat outdated and the next version SQ 6.5 will be released early August.

**Comment by guwirth** [comment]
Hello @Bertk, https://blog.travis-ci.com/2017-07-11-trusty-as-default-linux-is-coming is doing the trick. Please have a look at my last pull request to adapt the travis.yml file. Regards

**Comment by guwirth** [comment]
@Bertk ready for final review?

**Comment by Bertk** [comment]
@guwirth Should be OK now.

**Comment by guwirth** [comment]
@jmecosta do you also wanna do a review or test it?

**Pull request merged**